packages/shared/src/hooks/useFeedSettings.ts

```diff
@@ -76,11 +83,20 @@ export default function useFeedSettings(): FeedSettingsReturnType {
   const { data: feedQuery = {}, isLoading } = useQuery<AllTagCategoriesData>(
     filtersKey,
     async () => {
-      const req = await request(`${apiUrl}/graphql`, FEED_SETTINGS_QUERY, {
-        loggedIn: !!user?.id,
-      });
-
-      return { ...feedQuery, ...req };
+      const req = await request<AllTagCategoriesData>(
+        `${apiUrl}/graphql`,
+        FEED_SETTINGS_QUERY,
+        { loggedIn: !!user?.id },
+      );
+
+      const feedSettings = getQueryFeedSettings(
+        req,
+        feedQuery,
+        !!user,
+        shouldShowMyFeed,
+      );
```

Comment:
I think you should run the request only when it is needed.
Right now you dispatch the request and then check if you need it inside the `getQueryFeedSettings`

Reply:
ahh totally makes sense, less processing.

Reply:
Though we would really still need this request as this contains the list of tags/keywords.

We will now only decide whether we will utilize remote feed settings.

Reply:
Oh. Sorry I missed it. So it makes sense to keep it this way